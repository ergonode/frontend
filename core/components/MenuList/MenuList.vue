/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="menu-section">
        <MenuListHeader
            v-if="sectionTitle"
            :title="sectionTitle" />
        <ul class="menu-section__list">
            <MenuListElement
                v-for="(item, index) in sectionMenu"
                :key="index"
                :item="item" />
        </ul>
    </div>
</template>

<script>
export default {
    name: 'MenuList',
    components: {
        MenuListElement: () => import('~/core/components/MenuList/MenuListElement'),
        MenuListHeader: () => import('~/core/components/MenuList/MenuListHeader'),
    },
    props: {
        sectionTitle: {
            type: String,
            default: null,
        },
        sectionMenu: {
            type: Array,
            required: true,
        },
    },
};
</script>

<style lang="scss">
    @import "~/assets/scss/navigation/menu-list.scss";
</style>
