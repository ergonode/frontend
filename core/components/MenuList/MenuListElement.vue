/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <li
        :class="['list-element', { 'list-element--selected': isSelected }]"
        @mouseover="onMouseOver"
        @mouseout="onMouseOut">
        <NuxtLink
            class="list-element__link"
            :to="{ path: item.routing }">
            <Component
                :is="iconComponent"
                :fill-color="iconFillColor" />
            <span
                class="link-text font--medium-16-24"
                v-text="item.title" />
        </NuxtLink>
    </li>
</template>

<script>
import { GRAPHITE, GREEN, WHITE } from '~/assets/scss/_variables/_colors.scss';

export default {
    name: 'MenuListElement',
    props: {
        item: {
            type: Object,
            required: true,
        },
    },
    data() {
        return {
            isHover: false,
        };
    },
    computed: {
        iconFillColor() {
            let iconColor = GRAPHITE;
            if (this.isHover) iconColor = GREEN;
            if (this.isSelected) iconColor = WHITE;

            return iconColor;
        },
        iconComponent() {
            return () => import(`~/components/Icon/Menu/Icon${this.item.icon}`);
        },
        isSelected() {
            return this.$route.path.includes(this.item.routing);
        },
    },
    methods: {
        onMouseOver() {
            if (!this.isSelected) this.isHover = true;
        },
        onMouseOut() {
            this.isHover = false;
        },
    },

};
</script>
