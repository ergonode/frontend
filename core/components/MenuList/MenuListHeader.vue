/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template functional>
    <h3
        class="menu-section__title font--medium-12-16"
        v-text="props.title" />
</template>

<script>
export default {
    name: 'MenuListHeader',
    props: {
        title: {
            type: String,
            default: '',
        },
    },
};
</script>
