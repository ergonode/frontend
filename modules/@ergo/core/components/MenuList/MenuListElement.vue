/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <li
        :class="['list-element', { 'list-element--selected': isSelected }]"
        @mouseover="onMouseOver"
        @mouseout="onMouseOut">
        <NuxtLink
            class="list-element__link"
            :to="{ path: item.routing }">
            <Component
                :is="iconComponent"
                :fill-color="iconFillColor" />
            <span
                class="link-text"
                v-text="item.title" />
        </NuxtLink>
    </li>
</template>

<script>
import { GRAPHITE, GREEN, WHITE } from '@Core/assets/scss/_js-variables/colors.scss';

export default {
    name: 'MenuListElement',
    props: {
        item: {
            type: Object,
            required: true,
        },
    },
    data() {
        return {
            isHover: false,
        };
    },
    computed: {
        iconFillColor() {
            let iconColor = GRAPHITE;
            if (this.isHover) iconColor = GREEN;
            if (this.isSelected) iconColor = WHITE;

            return iconColor;
        },
        iconComponent() {
            return () => import(`@Core/components/Icons/Menu/Icon${this.item.icon}`);
        },
        isSelected() {
            return this.$route.path.includes(this.item.routing);
        },
    },
    methods: {
        onMouseOver() {
            if (!this.isSelected) this.isHover = true;
        },
        onMouseOut() {
            this.isHover = false;
        },
    },

};
</script>
