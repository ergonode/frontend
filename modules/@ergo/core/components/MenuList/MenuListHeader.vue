/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template functional>
    <h3
        class="menu-section__title"
        v-text="props.title" />
</template>

<script>
export default {
    name: 'MenuListHeader',
    props: {
        title: {
            type: String,
            default: '',
        },
    },
};
</script>
