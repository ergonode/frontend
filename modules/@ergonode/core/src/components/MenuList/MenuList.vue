/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="menu-section">
        <MenuListHeader
            v-if="sectionTitle"
            :title="sectionTitle" />
        <ul class="menu-section__list">
            <MenuListElement
                v-for="(item, index) in sectionMenu"
                :key="index"
                :item="item" />
        </ul>
    </div>
</template>

<script>
export default {
    name: 'MenuList',
    components: {
        MenuListElement: () => import('@Core/components/MenuList/MenuListElement'),
        MenuListHeader: () => import('@Core/components/MenuList/MenuListHeader'),
    },
    props: {
        sectionTitle: {
            type: String,
            default: null,
        },
        sectionMenu: {
            type: Array,
            required: true,
        },
    },
};
</script>

<style lang="scss">
    @import "@Core/assets/scss/navigation/menu-list.scss";
</style>
