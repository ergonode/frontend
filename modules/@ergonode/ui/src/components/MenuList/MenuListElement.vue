/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <li
        :class="classes"
        @mouseover="onMouseOver"
        @mouseout="onMouseOut">
        <NuxtLink
            class="menu-list-element__link"
            :to="{ path: item.routing }">
            <Component
                :is="iconComponent"
                :fill-color="iconFillColor" />
            <span
                class="link-text"
                v-text="item.title" />
        </NuxtLink>
    </li>
</template>

<script>
import {
    GRAPHITE,
    GREEN,
    WHITE,
} from '@UI/assets/scss/_js-variables/colors.scss';

export default {
    name: 'MenuListElement',
    props: {
        /**
         * Item data model
         */
        item: {
            type: Object,
            required: true,
        },
    },
    data() {
        return {
            isHover: false,
        };
    },
    computed: {
        classes() {
            return [
                'menu-list-element',
                {
                    'menu-list-element--selected': this.isSelected,
                },
            ];
        },
        iconFillColor() {
            let iconColor = GRAPHITE;
            if (this.isHover) iconColor = GREEN;
            if (this.isSelected) iconColor = WHITE;

            return iconColor;
        },
        iconComponent() {
            return () => import(`@Core/components/Icons/Menu/Icon${this.item.icon}`);
        },
        isSelected() {
            return this.$route.path.includes(this.item.routing);
        },
    },
    methods: {
        onMouseOver() {
            if (!this.isSelected) this.isHover = true;
        },
        onMouseOut() {
            this.isHover = false;
        },
    },

};
</script>

<style lang="scss" scoped>
    .menu-list-element {
        position: relative;
        z-index: $Z_INDEX_LVL_0;
        width: 100%;
        height: 48px;
        background-color: $WHITE;
        cursor: pointer;

        &:not(&--selected) {
            &:hover {
                z-index: $Z_INDEX_NAV;
                background-color: $WHITESMOKE;
                box-shadow: $ELEVATOR_2_DP;

                span {
                    color: $GRAPHITE_DARK;
                }
            }
        }

        &__link {
            display: flex;
            align-items: center;
            height: 100%;
            text-decoration: none;
            padding: 0 24px;

            & .link-text {
                margin-left: 24px;
                color: $GRAPHITE;
                font: $FONT_MEDIUM_16_24;
                text-align: left;
                text-overflow: ellipsis;
                overflow: hidden;
                word-break: break-all;
            }
        }

        &--selected {
            background-color: $GREEN;
            color: $WHITE;

            .link-text {
                color: $WHITE;
            }
        }
    }
</style>
