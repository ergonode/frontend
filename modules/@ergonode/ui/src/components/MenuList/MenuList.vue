/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="menu-section">
        <MenuListHeader
            v-if="title"
            :title="title" />
        <ul class="menu-section__list">
            <MenuListElement
                v-for="(item, index) in menu"
                :key="index"
                :item="item" />
        </ul>
    </div>
</template>

<script>
import MenuListElement from '@UI/components/MenuList/MenuListElement';
import MenuListHeader from '@UI/components/MenuList/MenuListHeader';

export default {
    name: 'MenuList',
    components: {
        MenuListElement,
        MenuListHeader,
    },
    props: {
        /**
         * The title of the component
         */
        title: {
            type: String,
            default: null,
        },
        /**
         * List of menu items
         */
        menu: {
            type: Array,
            required: true,
        },
    },
};
</script>

<style lang="scss">
    .menu-section {
        &:not(:nth-child(1)) {
            border-top: 1px solid rgba( $GREY, 0.5 );
        }

        &__title {
            position: relative;
            padding: 12px 16px;
            color: $GREY_DARK;
            font: $FONT_MEDIUM_12_16;
            text-align: left;
            text-transform: uppercase;
            letter-spacing: 0.5px;

            &::after {
                position: absolute;
                top: 0;
                left: 40px;
                width: 34px;
                height: 41px;
                background: linear-gradient(90deg, rgba(255, 255, 255, 0) 0%, $WHITE 100%);
                transition: opacity 0.9s;
                opacity: 0;
                content: "";
            }
        }

        &__list {
            display: flex;
            flex-wrap: wrap;
            justify-content: flex-start;
            align-items: center;
            width: 100%;
            padding: 0;
            list-style: none;
            overflow: hidden;
        }
    }
</style>
