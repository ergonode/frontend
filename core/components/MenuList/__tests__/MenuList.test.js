/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import { shallowMount, createLocalVue } from '@vue/test-utils';
// eslint-disable-next-line import/no-extraneous-dependencies
import VueRouter from 'vue-router';
import MenuList from '~/core/components/MenuList/MenuList';

const localVue = createLocalVue();
const router = new VueRouter();
const mocks = {
    $route: {
        path: '/products',
    },
};

describe('Navigation/MenuList', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(MenuList, {
            localVue,
            router,
            mocks,
            propsData: {
                sectionMenu: [
                    {
                        title: 'Products catalog',
                        routing: '/products',
                        icon: 'Document',
                    },
                ],
            },
            stubs: ['NuxtLink'],
        });
    });

    it('Component is named well', () => {
        expect(typeof MenuList.name).toBe('string');
        expect(MenuList.name).toEqual('MenuList');
        expect(wrapper).toMatchSnapshot();
    });
});
