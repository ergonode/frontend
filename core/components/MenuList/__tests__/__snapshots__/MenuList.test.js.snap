// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/MenuList Component is named well 1`] = `
<div class="menu-section">
  <!---->
  <ul class="menu-section__list">
    <li class="list-element list-element--selected">
      <nuxtlink-stub to="[object Object]" class="list-element__link">
        <baseicon-stub iconname="box" width="24" height="24" iconcolor="currentColor">
          <path d="M14,2 L6,2 C4.9,2 4.01,2.9 4.01,4 L4,20 C4,21.1 4.89,22 5.99,22 L18,22 C19.1,22 20,21.1 20,20 L20,8 L14,2 L14,2 Z M16,18 L8,18 L8,16 L16,16 L16,18 L16,18 Z M16,14 L8,14 L8,12 L16,12 L16,14 L16,14 Z M13,9 L13,3.5 L18.5,9 L13,9 L13,9 Z"></path>
        </baseicon-stub> <span class="link-text font--medium-16-24">Products catalog</span>
      </nuxtlink-stub>
    </li>
  </ul>
</div>
`;
